### PR TITLE
refactor(export): flexible exporters generation

### PR DESCRIPTION
**Pull request by danilshvalov**
Hi! Thank you for this awesome plugin!

I found that the interface of default exporters is not very flexible, so I decided to improve it

- It's easy to add a new exporter. For example, I added the `pandoc` exporter for markdown:

```lua
submenu('m', 'Export to Markdown file', 'md', {
    emacs = {
        command = 'org-md-export-to-markdown'
    },
    pandoc = {
        format = "gfm"
    }
}),
```
- The labels of the exporters is generated automatically. For example, `My awesome exporter` becomes `My awesome exporter (emacs/pandoc)`:

```lua
submenu('d', 'My awesome exporter', 'docx', {
    emacs = {
        -- ...
    },
    pandoc = {
        -- ...
    }
}),
```

- The arguments for the exporters were separated

### CODE REVIEW

File: lua/orgmode/export/init.lua [hunk 1]
```diff
@@ -172,4 +47,163 @@ function Export._exporter(cmd, target, on_success, on_error)
...
+    }),
+    submenu('m', 'Export to Markdown file', 'md', {
+      emacs = {
+        command = 'org-md-export-to-markdown',
+      },
+      pandoc = {
+        format = 'gfm',
+      },
+    }),
+    submenu('i', 'Export to iCalendar file', 'ics', {
```
Comment (kristijanhusak) [inline lua/orgmode/export/init.lua:184]: This shouldn't be a submenu since it has only 1 option.

**Review by kristijanhusak — CHANGES_REQUESTED** [review]
Thanks for the PR!

I agree this is much better. Just please address the iCalendar thing and this is good to go!

File: lua/orgmode/export/init.lua (hunk 1 above)
Comment (danilshvalov) [reply lua/orgmode/export/init.lua:191]: Done

**Review by kristijanhusak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**